dynamic-proxy/src/main/java/com/iluwatar/dynamicproxy/tinyrestclient/TinyRestClient.java

```diff
@@ -0,0 +1,157 @@
+    var bodyPublisher = buildBodyPublisher(method, args);
+    var httpRequest = HttpRequest.newBuilder()
+        .uri(URI.create(url))
+        .header("Content-Type", "application/json")
+        .method(httpMethodName, bodyPublisher)
+        .build();
+    var httpResponse = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+    return getResponse(method, httpResponse);
+  }
+
+  private String buildUrl(Method method, Object[] args, Annotation httpMethodAnnotation) {
+    var url = annotationValue(httpMethodAnnotation);
+    if (url == null) {
+      return "";
+    }
+    var index = 0;
+    for (var parameter : method.getParameters()) {
+      var pathAnnotation = getAnnotationOf(parameter.getDeclaredAnnotations(), Path.class);
+      if (pathAnnotation != null) {
+        var pathParam = "{" + annotationValue(pathAnnotation) + "}";
+        var pathValue = args[index].toString();
+        url = url.replace(pathParam, pathValue);
+      }
+      index++;
+    }
+    return url;
+  }
+
+  private HttpRequest.BodyPublisher buildBodyPublisher(Method method, Object[] args) {
+    var index = 0;
+    for (var parameter : method.getParameters()) {
+      var bodyAnnotation = getAnnotationOf(parameter.getDeclaredAnnotations(), Body.class);
+      if (bodyAnnotation != null) {
+        var body = JsonUtil.objectToJson(args[index]);
+        return HttpRequest.BodyPublishers.ofString(body);
+      }
+      index++;
+    }
+    return HttpRequest.BodyPublishers.noBody();
+  }
+
+  private Object getResponse(Method method, HttpResponse<String> httpResponse) {
+    var rawData = httpResponse.body();
+    var returnType = method.getGenericReturnType();
+    if (returnType instanceof ParameterizedType) {
+      Class<?> responseClass = (Class<?>) (((ParameterizedType) returnType)
+          .getActualTypeArguments()[0]);
+      return JsonUtil.jsonToList(rawData, responseClass);
+    } else {
+      Class<?> responseClass = method.getReturnType();
+      return JsonUtil.jsonToObject(rawData, responseClass);
+    }
+  }
+
+  private Annotation getHttpMethodAnnotation(Annotation[] annotations) {
+    return Arrays.stream(annotations)
+        .filter(annot -> annot.annotationType().isAnnotationPresent(HttpMethod.class))
+        .findFirst().orElse(null);
+  }
+
+  private Annotation getAnnotationOf(Annotation[] annotations, Class<?> clazz) {
+    return Arrays.stream(annotations)
+        .filter(annot -> annot.annotationType().equals(clazz))
+        .findFirst().orElse(null);
+  }
+
+  private String annotationValue(Annotation annotation) {
+    var valueMethod = Arrays.stream(annotation.annotationType().getDeclaredMethods())
+        .filter(methodAnnot -> methodAnnot.getName().equals("value"))
+        .findFirst().orElse(null);
+    if (valueMethod == null) {
+      return null;
+    }
+    Object result;
+    try {
+      result = valueMethod.invoke(annotation, (Object[]) null);
+    } catch (Exception e) {
```

Comment:
Consider logging this exception or rethrowing as a custom unchecked exception for better debugging.

Reply:
Done. Logging the exception.